catalyst/metrics/funtional_metric.py

```diff
@@ -0,0 +1,97 @@
+from typing import Callable, Dict
+
+import torch
+
+from catalyst.metrics import ICallbackBatchMetric
+
+
+class BatchFuntionalMetric(ICallbackBatchMetric):
+    """
+    Class for custom metric in functional way.
+    Note: the loader metrics calculated as average over all examples
+    """
+
+    def __init__(
+        self, metric_function: Callable, prefix: str,
+    ):
+        """
+
+        Args:
+            metric_function: metric function, that get outputs, targets and return score as
+            torch.Tensor
+            prefix: metric prefix
+        """
+        super().__init__(compute_on_call=True, prefix=prefix)
+        self.metric_function = metric_function
+        self.cumulative_metric = None
+        self.total_examples = 0
+
+    def reset(self):
+        """Reset all statistics"""
+        self.cumulative_metric = None
+        self.total_examples = 0
+
+    def _update_cumulative_metric(self, value: torch.Tensor, batch_length: int) -> None:
+        if self.cumulative_metric is None:
+            self.cumulative_metric = value
+            self.total_examples = batch_length
+        else:
+            old = (
+                self.total_examples / (self.total_examples + batch_length) * self.cumulative_metric
+            )
+            new = batch_length / (self.total_examples + batch_length) * value
+            self.cumulative_metric = old + new
+            self.total_examples += batch_length
```

Comment:
could we use https://github.com/catalyst-team/catalyst/blob/dev/catalyst/metrics/_additive.py#L8 here?

Reply:
like https://github.com/catalyst-team/catalyst/blob/dev/catalyst/metrics/_accuracy.py#L61